clients/agentic-sandbox-client-python/agentic_sandbox/sandbox.py

```diff
@@ -0,0 +1,188 @@
+            plural=SANDBOX_PLURAL_NAME,
+            field_selector=f"metadata.name={self.claim_name}",
+            timeout_seconds=180
+        ):
+            sandbox_object = event['object']
+            status = sandbox_object.get('status', {})
+            conditions = status.get('conditions', [])
+            is_ready = False
+            for cond in conditions:
+                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                    is_ready = True
+                    break
+
+            if is_ready:
+                self.sandbox_name = sandbox_object['metadata']['name']
+                w.stop()
+                print(f"Sandbox {self.sandbox_name} is ready.")
+                break
+
+        if not self.sandbox_name:
+            self.__exit__(None, None, None)
+            raise TimeoutError("Sandbox did not become ready within the 180-second timeout period.")
+
+        print(f"Starting port-forwarding for sandbox {self.sandbox_name}...")
+        self.port_forward_process = subprocess.Popen(
+            [
+                "kubectl", "port-forward",
+                f"pod/{self.sandbox_name}",
+                f"{self.server_port}:{self.server_port}",
+                "-n", self.namespace
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        time.sleep(3)  # Give port-forward a moment to establish
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Deletes the SandboxClaim resource and stops port-forwarding."""
+        if self.port_forward_process:
+            print("Stopping port-forwarding...")
+            self.port_forward_process.terminate()
+            self.port_forward_process.wait()
+
+        if self.claim_name:
+            print(f"Deleting SandboxClaim: {self.claim_name}")
+            try:
+                self.custom_objects_api.delete_namespaced_custom_object(
+                    group=CLAIM_API_GROUP,
+                    version=CLAIM_API_VERSION,
+                    namespace=self.namespace,
+                    plural=CLAIM_PLURAL_NAME,
+                    name=self.claim_name
+                )
+            except client.ApiException as e:
+                if e.status != 404:
+                    print(f"Error deleting sandbox claim: {e}")
+
+    def run(self, command: str, timeout: int = 60) -> ExecutionResult:
+        """Executes a shell command inside the running sandbox."""
+        if not self.is_ready():
+            raise ConnectionError("Sandbox is not ready. Cannot execute commands.")
+
+        url = f"http://127.0.0.1:{self.server_port}/execute"
+        payload = {"command": command}
+
+        response = requests.post(url, json=payload, timeout=timeout)
+        response.raise_for_status()
+
+        response_data = response.json()
+        return ExecutionResult(
+            stdout=response_data['stdout'],
+            stderr=response_data['stderr'],
```

Comment:
The `path` parameter in the `write` method is used to get the `filename`, but the directory part of the path is ignored. This might be confusing for users. It would be better to clarify in the docstring that only the basename of the path is used as the filename in the sandbox.